### PR TITLE
C 2 2

### PR DESCRIPTION
**Pull request by KorzhovIgor**
was added new routes for updating, new commands such as migrate and seed, also was added MVC pattern.

### CODE REVIEW

**Review by VitaliGolubka — APPROVED** [review]

File: app/controllers/AppController.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+class AppController
+{
+
+    public function index()
+    {
+        return require_once PATH_TO_PROJECT . '/app/views/index.html';
+    }
+
+    public function error()
```
[the remark this answers is not in the document]
Comment (KorzhovIgor) [reply app/controllers/AppController.php]: done

File: app/controllers/CommentsController.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    public function show($id)
+    {
+        $comment = json_decode($this->commentsModel->getByID($id));
+
+        return require_once PATH_TO_PROJECT . '/app/views/Comments/show.php';
+    }
+
+    public function create()
```
[the remark this answers is not in the document]
Comment (KorzhovIgor) [reply app/controllers/CommentsController.php]: done

File: app/controllers/CommentsController.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        return require_once PATH_TO_PROJECT . '/app/views/Comments/create.php';
+    }
+
+    public function store()
+    {
+        if ((strlen($_POST['title']) > 0) || (strlen($_POST['content']) > 5)) {
```
[the remark this answers is not in the document]
Comment (KorzhovIgor) [reply app/controllers/CommentsController.php]: done was created new folder Request and new class for it

File: app/models/Migration.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            'created_at' => date("Y-m-d"),
+        ];
+        $sql = <<<SQL
+            INSERT INTO migrations (name, created_at) 
+            VALUES (:name, :created_at)
+        SQL;
+        $preparedRequest = self::$connection->prepare($sql);
+        $preparedRequest->execute($params);
+    }
+}
```
[the remark this answers is not in the document]
Comment (KorzhovIgor) [reply app/models/Migration.php]: done

File: app/views/Comments/edit.php [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        </div>
+        <textarea name="content" id="content" cols="30" rows="10" required><?=$comment->content ?></textarea>
+        <div>
+            <button id="submitComment" type="submit">Submit</button>
+        </div>
+    </form>
+</div>
+<script src="/assets/javascript/form.js"></script>
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (KorzhovIgor) [reply app/views/Comments/edit.php]: done

File: app/views/Comments/show.php [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        <h1>Show page</h1>
+        <h1>ID: <?= $comment->id ?></h1>
+        <h1>Title: <?= $comment->title ?></h1>
+        <h1>Content: <?= $comment->content ?></h1>
+        <h1>Created: <?= $comment->created_at ?></h1>
+    </div>
+</div>
+<script src=""></script>
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (KorzhovIgor) [reply app/views/Comments/show.php]: done

File: app/views/error.html [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Error</title>
+</head>
+<body>
+<h1>Error</h1>
+</body>
+</html>
```
[the remark this answers is not in the document]
Comment (KorzhovIgor) [reply app/views/error.html]: done

File: commands/migrate.php [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    $allFiles = glob(PATH_TO_MIGRATIONS . '*.sql');
+    migrate($migrationModel, $connection, $allFiles);
+}
+
+function migrate(Migration $migrationModel, PDO $connection, array $allFiles)
+{
+    if (checkExistTable($connection, 'migrations')) {
+        $migrations = $migrationModel->getAll();
+        foreach ($allFiles as $file) {
+            $a = 0;
```
[the remark this answers is not in the document]
Comment (KorzhovIgor) [reply commands/migrate.php]: fixed

**Pull request merged**